docs/catalog/web/minimize-main-thread-work.md

```diff
@@ -0,0 +1,42 @@
+---
+version: 1.0
+submitted_by: markus-ntt-seidl
+published_date: TBD
+category: web
+tags: 
+ - web
+ - role:web-developer
+ - size:small
+---
+
+# Minimize main thread work
+
+## Description
+
+Web browsers traditionally consist of a main rendering thread that handles most of the updates on the web page and the execution of JavaScript. JavaScript is executed on the main thread to simplify JavaScript implementations so the JavaScript programmer doesn't have to deal with any multi-threading programming patterns.
+When doing long running JavaScript computations it is running single threaded. In contrast all CPUs nowadays have multiple cores which have to be powered on while being underutilized.
+
+## Solution
+
+For long running JavaScript computations try to use WebWorkers and run them in another thread while keeping the main rendering thread free. Consider moving the JavaScript computation to efficient server implementations that use optimized algorithms.
```

Comment:
Do you have any guidance to add here about what defines "long running"?

Reply:
I found this: 
![image](https://user-images.githubusercontent.com/103509715/201008905-36f65332-a10e-4eee-8ac9-d012286b4dc7.png)
Which is Total Blocking Time (should TBT instead of TTB in the header), you get better scores when you are <100ms .

Reply:
Sweet!  Can you add this into the section or as a reference for others to use?